### PR TITLE
Fix linebreaking in long binary op chains

### LINKED ISSUE #1035: edit + slurp roundtrip fails for long lines with repeated operators
The codebase slurps this in fine:
```
foo : Text
foo =
  "aaaaaaaaaaaaaaaaaaaaaa" ++ "bbbbbbbbbbbbbbbbbbbbbb" ++ "cccccccccccccccccccccc"
```

But it pretty prints it as:
```
foo : Text
foo =
  use Text ++
     "aaaaaaaaaaaaaaaaaaaaaa"
  ++ "bbbbbbbbbbbbbbbbbbbbbb"
  ++ "cccccccccccccccccccccc"
```

And that won't go back in:
```
  /home/traveller/code/mine/unison-ian/scratch.u:5:3:
  unexpected ++
  expecting [, bang, binding, case, false, handle, if, lambda, let, namespace, quote, termLink, true, tuple, or typeLink
      5 |   ++ "bbbbbbbbbbbbbbbbbbbbbb"
```

(This is with `ucm version: release/M2-base-329-gd3611ffd`)

### PR DESCRIPTION
**Pull request by runarorama**
This changes the way the term printer splits repeated binary operations into columns, which was just wrong.

Before: 

```
foo : Text
foo =
  use Text ++
     "aaaaaaaaaaaaaaaaaaaaaa"
  ++ "bbbbbbbbbbbbbbbbbbbbbb"
  ++ "cccccccccccccccccccccc"
```

After : 

```
foo : Text
foo =
  use Text ++
  "aaaaaaaaaaaaaaaaaaaaaa"
    ++ "bbbbbbbbbbbbbbbbbbbbbb"
    ++ "cccccccccccccccccccccc"
```
Fixes #1035 and #2165.

### CODE REVIEW

**Comment by hojberg** [comment]
Would it be possible to keep the operator on the front, but indent the lines after the first?

so:
```
foo : Text
foo =
   "aaaaaaaaaaaaaaaaaaaaaa" ++
     "bbbbbbbbbbbbbbbbbbbbbb" ++
     "cccccccccccccccccccccc"
```

would be

```
foo : Text
foo =
   "aaaaaaaaaaaaaaaaaaaaaa"
     ++ "bbbbbbbbbbbbbbbbbbbbbb"
     ++ "cccccccccccccccccccccc"
```

instead?

My main reasoning is that the `|>` operator, I think, would look strange at the end.
To me this:

```
foo : Text
foo =
  users
    |> List.map getFullName
    |> List.map capitalize
```

Reads a lot nicer than:

```
foo : Text
foo =
  users |>
    List.map getFullName |>
    List.map capitalize
```

(This is totally something we could do later)

**Comment by hojberg** [comment]
I'm pretty excited to see this fixed! this is probably the pretty printing bug I hit most often 🎉

**Comment by runarorama** [comment]
Yeah, we can do whatever we want

**Comment by runarorama** [comment]
Done and done

**Review by pchiusano — APPROVED** [review]
Sweet! Looks like just some CI failures though.

**Comment by pchiusano** [comment]
I don't know what's going on but after I approved it looks like it allows me to merge, even though CI is failing. I removed the ready-to-merge label just in case since I don't want it automatically merging something that doesn't pass CI.

I think this change is good to go though once CI passes.

**Pull request merged**
